djautotask/sync_rest.py

```diff
@@ -271,3 +293,134 @@ def _assign_field_data(self, instance, json_data):
     def get_page(self, next_url=None, *args, **kwargs):
         kwargs['conditions'] = self.api_conditions
         return self.client.get_contacts(next_url, *args, **kwargs)
+
+
+# ParentSynchronizer: using SOAP API
+class TicketSynchronizer(SyncRecordUDFMixin, Synchronizer, ParentSynchronizer):
```

Comment:
I think I'm seeing a problem with looking up UDF records. I get

```
[2021-04-24 13:19:55] DEBUG [djautotask.sync:319] No UDF records returned for Name: TicketUDF mat
ching query does not exist.                                                                      
[2021-04-24 13:19:55] DEBUG [djautotask.sync:319] No UDF records returned for Name: TicketUDF mat
ching query does not exist.                                                                      
[2021-04-24 13:19:55] DEBUG [kanban.autotask.models.ticketboard:586] Handling Ticket pre_save sig
nal for 9990.                                                                                    
[2021-04-24 13:19:55] INFO [kanban.autotask.models.ticketboard:565] Handling Ticket post_save sig
nal for 9990 (status is New).                                                                    
[2021-04-24 13:19:55] INFO [djautotask.sync_rest:197] Updated: Ticket 9990-Investigate 404       
Ticket Sync Summary - Created: 0, Updated: 1, Skipped: 0                                         
```
when syncing a ticket with UDF fields set. I'm thinking maybe the REST API stores UDF data differently and the current UDF mixin might need to be adjusted. 


Can you confirm if you are able to sync UDF info on tickets correctly?

Reply:
Oh, thanks for the test which I missed. I will look into that.

Reply:
https://git.ops.topleft.team/apps/django-topleft/-/issues/1798#note_28544

With permission modified and the most recent code, UDF is now working.

![image](https://user-images.githubusercontent.com/73486314/116267434-43f07400-a731-11eb-84dc-3ece69cbd1ef.png)

Thanks Cameron!

Reply:
That's good news! I thought we might've needed a lot more effort to get it working. Thanks for sorting that out Daniel.

Reply:
Ooops, I forgot to push!

Please confirm the format is the same as it was stored before. I have no idea and experience about picklist, so I am not sure it is correctly handled. Could you check about that? And could you let me know or show how to test picklist and what it is. Thanks Cemeron.

Reply:
No problem. 

UDFs that are picklists look like they are being displayed correctly. Picklists just specify some values/choices that can be chosen for a UDF field. Just like a choice field in a form.  For example, the Skill Level UDF field has a picklist of choices: Beginner, Intermediate, etc. 

 
![image](https://user-images.githubusercontent.com/44041923/116292831-013b9580-a74b-11eb-9bb4-874f504449c2.png)

To test that they are displaying, you can turn on the UDF fields in the board edit page. 

I did find one UDF type that needs to be updated. It looks like UDFs with date fields are now being sent as datetimes. Example: 
![image](https://user-images.githubusercontent.com/44041923/116293078-4eb80280-a74b-11eb-8f34-6d42534544ba.png)